src/mbgl/renderer/renderer.cpp

```diff
@@ -52,6 +52,17 @@ std::vector<Feature> Renderer::queryRenderedFeatures(const ScreenBox& box, const
             options
     );
 }
+    
+std::vector<Feature> Renderer::queryRenderedSourceFeatures(const ScreenBox& box) const {
```

Comment:
This method could be removed and it's logic encapsulated in `Renderer::queryShapeAnnotations`

Reply:
Technically both methods should return different annotation types with different parameters. What I did is refactor the logic that was common to both `queryPointAnnotations` and `queryShapeAnnotations`.

Reply:
I don't follow, `queryPointAnnotations` calls `queryRenderedFeatures`. 
Additionally, this method only calls `queryShapeAnnotations`, but has a name that implies it is a generic query for a source.